salt/states/chocolatey.py

```diff
@@ -262,3 +262,170 @@ def uninstalled(name, version=None, uninstall_args=None, override_args=False):
     ret['changes'] = salt.utils.data.compare_dicts(pre_uninstall, post_uninstall)
 
     return ret
+
+
+def upgraded(name,
+             version=None,
+             source=None,
+             force=False,
+             pre_versions=False,
+             install_args=None,
+             override_args=False,
+             force_x86=False,
+             package_args=None):
+    '''
+    Upgrades a package. Will install the package if not installed.
+
+    .. versionadded: Oxygen
+
+    Args:
+
+        name (str):
+            The name of the package to be installed. Required.
+
+        version (str):
+            Install a specific version of the package. Defaults to latest
+            version. If the version is greater than the one installed then the
+            specified version will be installed. Default is ``None``.
+
+        source (str):
+            Chocolatey repository (directory, share or remote URL, feed).
+            Defaults to the official Chocolatey feed. Default is ``None``.
+
+        force (bool):
+            ``True`` will reinstall an existing package with the same version.
+            Default is ``False``.
+
+        pre_versions (bool):
+            ``True`` will nclude pre-release packages. Default is ``False``.
+
+        install_args (str):
+            Install arguments you want to pass to the installation process, i.e
+            product key or feature list. Default is ``None``.
+
+        override_args (bool):
+            ``True`` will override the original install arguments (for the
+            native installer) in the package and use those specified in
+            ``install_args``. ``False`` will append install_args to the end of
+            the default arguments. Default is ``False``.
+
+        force_x86 (bool):
+            ``True`` forces 32bit installation on 64 bit systems. Default is
+            ``False``.
+
+        package_args (str):
+            Arguments you want to pass to the package. Default is ``None``.
+
+    .. code-block:: yaml
+
+        upgrade_some_package:
+          chocolatey.upgraded:
+            - name: packagename
+            - version: '12.04'
+            - source: 'mychocolatey/source'
```

Comment:
Add a Note along the lines of.   This chocolatey pulls data from the Chocolatey Internet DB to perform this task and normal this is a slow operation, it may be faster with a local chocolatey repo.